cocotb/utils.py

```diff
@@ -118,7 +118,7 @@ def get_sim_steps(time, units=None):
         :exc:`ValueError`: If given *time* cannot be represented by simulator precision.
     """
     result = time
-    if units is not None:
+    if units is not None and units != "step":
```

Comment:
This is the typical idiom. 
```suggestion
    if units not in (None, 'step'):
```
One could also do a default check to correct `None` to `'step'` and only do the singular check  of `'step'` thereafter.